edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/Agent.cs

```diff
@@ -145,6 +145,10 @@ public async Task ReconcileAsync(CancellationToken token)
                                 throw;
                             }
                         }
+                        else
```

Comment:
Maybe flip the `if` and `else` around to simplify the condition. `if (plan.IsEmpty) { ... } else { ... }`